packages/base/src/formbuilder/objectform/geojsonsource.ts

```diff
@@ -0,0 +1,85 @@
+import { Ajv, ValidateFunction } from 'ajv';
+import * as geojson from 'geojson-schema/GeoJSON.json';
+
+import { BaseForm, IBaseFormProps } from './baseform';
+import { IDict } from '@jupytergis/schema';
+import { ISubmitEvent } from '@rjsf/core';
+
+/**
+ * The form to modify a GeoJSON source.
+ */
+export class GeoJSONSourcePropertiesForm extends BaseForm {
+  constructor(props: IBaseFormProps) {
+    super(props);
+    const ajv = new Ajv();
+    this._validate = ajv.compile(geojson);
+  }
+
+  protected processSchema(
+    data: IDict<any> | undefined,
+    schema: IDict,
+    uiSchema: IDict
+  ) {
+    super.processSchema(data, schema, uiSchema);
+    if (!schema.properties || !data) {
+      return;
+    }
+
+    // This is not user-editable
+    delete schema.properties.valid;
+
+    if (data.path !== '') {
+      this.removeFormEntry('data', data, schema, uiSchema);
+    }
+  }
+
+  protected onFormBlur(id: string, value: any) {
+    // Is there a better way to spot the path text entry?
+    if (!id.endsWith('_path')) {
+      return;
+    }
+
+    this._validatePath(value);
+  }
+
+  protected onFormSubmit = (e: ISubmitEvent<any>) => {
+    if (this.state.extraErrors?.path?.__errors?.includes('Invalid path')) {
+      return;
+    }
```

Comment:
It seems it is not triggered anymore, we can validate with an invalid path and nothing happens

Reply:
[According to the rjsf docs](https://rjsf-team.github.io/react-jsonschema-form/docs/api-reference/form-props/#extraerrors), it seems to be the expected behavior:

> By default, these are non-blocking errors, meaning that you can still submit the form when these are the only errors displayed to the user

Reply:
Looking at [the "Validation" section of the docs](https://rjsf-team.github.io/react-jsonschema-form/docs/usage/validation#async-validation):

> An important note is that these errors are "display only" and will not block the user from submitting the form again.

We probably don't have a choice here.

So let's not provide a text input but a file selector widget (using the FileBrowser widget from jupyterlab). That way we are sure the value is valid.

Reply:
Yes, it was the intent of the function above, to prevent validation if the path is invalid, by checking for a specific `extraError`.
Pretty sure it uses to work, but you're right, it should be fixed anyway by adding a filebrowser widget.

Reply:
Ah got it, I'll look into it

Reply:
I updated the code, it now shows an error dialog if the file is not valid, and it does not add the source upon submit